src/onegov/search/upgrade.py

```diff
@@ -3,28 +3,24 @@
 
 """
 from __future__ import annotations
+from uuid import uuid4
 
-from sqlalchemy import Column
-from sqlalchemy.dialects.postgresql import TSVECTOR
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.dialects.postgresql import TSVECTOR, JSONB
 
+from onegov.core.orm.types import UUID
 from onegov.core.upgrade import upgrade_task, UpgradeContext
-from onegov.search.utils import searchable_sqlalchemy_models
 
 
-@upgrade_task('Adds fts index column to all searchable models')
-def adds_index_data_columns(context: UpgradeContext) -> None:
-    models = (
-        model
-        for base in context.app.session_manager.bases
-        for model in searchable_sqlalchemy_models(base)
-    )
-    for model in models:
-        assert hasattr(model, '__tablename__')
-        if not context.has_column(model.__tablename__, 'fts_idx'):
-            context.operations.add_column(
-                model.__tablename__,
-                Column(
-                    'fts_idx',
-                    TSVECTOR,
-                )
-            )
+@upgrade_task('fts: Adds search index table to each schema')
+def adds_search_index_table(context: UpgradeContext) -> None:
+    if not context.has_table('search_index'):
+        context.operations.create_table(
+            'search_index',
+            Column('id', UUID, primary_key=True, default=uuid4),
+            Column('owner_id_int', Integer, nullable=True),
+            Column('owner_id_uuid', UUID, nullable=True),
+            Column('owner_type', String, nullable=False),
+            Column('fts_idx_data', JSONB, default={}),
+            Column('fts_idx', TSVECTOR)
+        )
```

Comment:
This migration is not necessary. The ORM automatically adds missing tables. You only need migrations for changes to the table columns/indexes/constraints.